inc/class.admin.manage.php

```diff
@@ -543,101 +559,177 @@ class="removeterm_type_matched_only" name="removeterm_type"
      * @return boolean
      * @author olatechpro
      */
-    public static function mergeTerms($taxonomy = 'post_tag', $old = '', $new = '')
+    public static function mergeTerms($taxonomy = 'post_tag', $old = '', $new = '', $merge_type = '')
     {
-        if (trim(str_replace(',', '', stripslashes($new))) == '') {
-            add_settings_error(__CLASS__, __CLASS__, esc_html__('No new term specified!', 'simple-tags'), 'error');
+        if ($merge_type === 'same_name'){
 
-            return false;
-        }
+            $old_terms = explode(',', $old);
 
-        // String to array
-        $old_terms = explode(',', $old);
-        $new_terms = explode(',', $new);
+            $old_terms = array_filter($old_terms, '_delete_empty_element');
 
-        // Remove empty element and trim
-        $old_terms = array_filter($old_terms, '_delete_empty_element');
-        $new_terms = array_filter($new_terms, '_delete_empty_element');
-        $common_elements = array_intersect($old_terms, $new_terms);
+            if (empty($old_terms)) {
+                add_settings_error(__CLASS__, __CLASS__, esc_html__('No terms provided for merging!', 'simple-tags'), 'error');
+                return false;
+            }
 
-        // If old/new tag are empty => exit !
-        if (empty($old_terms) || empty($new_terms)) {
-            add_settings_error(__CLASS__, __CLASS__, esc_html__('No new/old valid term specified!', 'simple-tags'), 'error');
+            $terms =[];
+            foreach ($old_terms as $term_name) {
+                $term_objects = get_terms([
+                    'taxonomy' => $taxonomy,
+                    'name' => sanitize_text_field($term_name),
+                    'hide_empty' => false,
+                ]);
 
-            return false;
-        }
+                if (is_array($term_objects) && count($term_objects) > 1) {
+                    $terms = array_merge($terms, $term_objects);
+                }
+            }
 
-        if (!empty($common_elements)) {
-            add_settings_error(__CLASS__, __CLASS__, esc_html__('Term to merge and New Term must not contain same term.', 'simple-tags'), 'error');
+            if (empty($terms)) {
+                add_settings_error(__CLASS__, __CLASS__, esc_html__('No terms with the same name found.', 'simple-tags'), 'error');
+                return false;
+            }
 
-            return false;
-        }
+            usort($terms, function ($a, $b) use ($term_name) {
+                // Score based on similarity to term name
+                similar_text($term_name, $a->slug, $similarity_a);
+                similar_text($term_name, $b->slug, $similarity_b);
```

Comment:
@Deji98 instead of using this function, why not say strtolower($term_a) == strtolower($term_b) to compare both terms at their lower case or isn't that working in your test?

I don't really see the reason for going through such length to compare if two strings are same thing when you can simply convert to same case and compare or is there a specific reason for going with your function?

Reply:
@ojopaul what I intend to do is to compare the retained term with the slugs so as to choose the most similar slug to the term name and It occurred to me that if terms have the same name but different slugs in the first place, the slugs were probably entered manually, we might have cases like slug-1, maybe something different from the term name or typos like slugg and nothing directly written to match the exact term name and in cases like that strtolower might not do the filtering properly.
So I decided to go with that so that It can choose the most similar slug to the term name

Reply:
@Deji98 In your current application, if terms has same name and different slug, are you not merging them into one?

Reply:
@ojopaul Yes I am merging them into one and retaining one of the slugs

Reply:
Or can you explain your logic for merging term so I can understand what the problem is. As something is certain, 2 terms under same taxonomy terms will never have same slug. So, if 2 terms shared same name, their slug will be different.

So, what's your current determination for merging terms with this function? @Deji98